### PR TITLE
Adds construction for wooden barricades, solar panels + solar trackers

### PR DESCRIPTION
**Pull request by Peptide90**
## About the PR 
Adds entities, construction graphs and textures for the wooden barricade, solar tracker and solar panel so that they can be built in-game and also be deconstructed.

I will continue working on construction graphs for remaining entities but wanted to put these out for now since they are usable.

Solar tracker currently doesn't work as it needs more stuff above my head but this gives someone a head start (Needs to be able to output solar angle and direction to all connected solar panels and to the solar control console).

Solar crate can be ordered at cargo to access 6 solar assembly parts. Will add glass to this crate when stack size can be specified. Solar tracker board isn't in as it doesn't function anyway but could be a protolathe recipe down the line. Solar assemblies could be researched autolathe recipe too but shall see.

Open to comments and feedback before this is merged.

**Screenshots**

barricadewoodensolar_assemblysolar_tracker

**Changelog**

:cl:
- add: Added construction of wooden barricades, solar panels & solar trackers (WIP)

### CODE REVIEW

File: Resources/Prototypes/Entities/Constructible/Power/power_base.yml [hunk 1]
```diff
@@ -268,3 +268,83 @@
   - type: ClientEntitySpawner
     prototypes:
     - HVDummyWire
+
+- type: entity
+  id: SolarAssembly
+  name: solar assembly
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Constructible/Power/power_base.yml:274]: i'd advise using new localization for the name and description, check Resources/Prototypes/Entities/Objects/Tools/gas_tanks.yml and Resources/Locale/en-US/entities/gas-tanks.ftl for examples

File: Resources/Prototypes/Entities/Constructible/Power/power_base.yml [hunk 2]
```diff
@@ -268,3 +268,83 @@
...
+        acts: ["Breakage"]
+  - type: Anchorable
+  - type: Pullable
+  - type: Construction
+    graph: solarpanel
+    node: solarassembly
+    
+- type: entity
+  id: SolarTracker
+  name: solar tracker
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Constructible/Power/power_base.yml:314]: same here

File: Resources/Prototypes/Entities/Constructible/Specific/barricades.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+- type: entity
+  id: Barricade
+  name: barricade
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Constructible/Specific/barricades.yml:3]: same here

File: Resources/Prototypes/Entities/Objects/Power/solar_parts.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿- type: entity
+  id: SolarAssemblyPart
+  name: Solar Assembly Part
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Power/solar_parts.yml:3]: same here (also, maybe shouldnt be a proper noun?)

File: Resources/Textures/Constructible/Structures/Doors/hidden_door.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,149 @@
+{
+  "version": 1,
```
Comment (mirrorcult) [inline Resources/Textures/Constructible/Structures/Doors/hidden_door.rsi/meta.json:2]: is the hidden door-stuff intended to be in this PR?

File: Resources/Prototypes/Entities/Objects/Power/solar_parts.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  - type: Item
+    size: 10
+  - type: Sprite
+    sprite: Objects/Power/solar_parts.rsi
+    state: solar_assembly_parts
+    
+- type: entity
+  id: SolarTrackerMachineCircuitboard
+  parent: BaseMachineCircuitboard
+  name: Solar Tracker (Machine Board)
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Power/solar_parts.yml:16]: same here (also, maybe shouldnt be a proper noun?)

File: Resources/Prototypes/Recipes/Construction/power.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  description: "Area Power Controller (APC). Controls power. In an area."
+  icon:
+    sprite: Constructible/Power/apc.rsi
+    state: apc0
+  objectType: Structure
+  placementMode: AlignWallProper
+  canBuildInImpassable: true
+  
+- type: construction
+  name: Solar Assembly
```
Comment (mirrorcult) [inline Resources/Prototypes/Recipes/Construction/power.yml:17]: here as well (though i dont see any examples for localizing construction yet)

File: Resources/Prototypes/Recipes/Construction/structures.yml [hunk 8]
```diff
@@ -198,16 +198,18 @@
   canBuildInImpassable: false
 
 - type: construction
-  name: APC
-  id: apc
-  graph: apc
+  name: Wooden Barricade
```
Comment (mirrorcult) [inline Resources/Prototypes/Recipes/Construction/structures.yml:201]: here as well (though i dont see any examples for localizing construction yet)

File: Resources/Textures/Constructible/Structures/Doors/hidden_door.rsi/meta.json (hunk 5 above)
Comment (Peptide90) [reply Resources/Textures/Constructible/Structures/Doors/hidden_door.rsi/meta.json:2]: I was going to include it in this PR and started the work but didn't finish...so still on the fence on finishing it off but the resources are there. Will get round to it.

Comment (Peptide90) [reply Resources/Textures/Constructible/Structures/Doors/hidden_door.rsi/meta.json]: Removed.

**Comment by Peptide90** [comment]
Changed to draft as need to change the name of the circuit board for the tracker. 

Will also look to change electronics boards into one yml file.

File: Resources/Textures/Objects/Power/solar_parts.rsi/meta.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+	"license": "CC-BY-SA-3.0",
+	"copyright": "Taken from https://tgstation13.org/wiki/Guide_to_construction#Solar_Panels_and_Trackers and modified.",
+    "states": [
+        {
+            "name": "solar_assembly_parts",
+
```
Comment (SweptWasTaken) [inline Resources/Textures/Objects/Power/solar_parts.rsi/meta.json:12]: Meta.json is badly indented here.

**Review by SweptWasTaken — CHANGES_REQUESTED** [review]
1 thing to fix.

**Comment by SweptWasTaken** [comment]
> 
> 
> Changed to draft as need to change the name of the circuit board for the tracker and won't have time till Monday.
> 
> Will also look to change electronics boards into one yml file.

I already organized circuit boards into their own yml file so just pull master. This PR looks good to me.

**Comment by Peptide90** [comment]
> 
> 
> 1 thing to fix.

Fixed indentations on meta.json and made remaining changes to things like materials on destruction so this is all tested on local and working now.

Only thing not up to par is the localisation which isn't working for the name or desctription for any of the entities or constructables added so they just show as item, their ID or blank in game (I also noticed oxygen tanks do this in the entity spawner too).

File: Resources/Textures/Constructible/Structures/barricades.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation",
```
Comment (DrSmugleaf) [inline Resources/Textures/Constructible/Structures/barricades.rsi/meta.json:4]: Missing commit hash here.

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: Resources/Textures/Constructible/Structures/barricades.rsi/meta.json (hunk 10 above)
Comment (SweptWasTaken) [reply Resources/Textures/Constructible/Structures/barricades.rsi/meta.json:4]: https://github.com/tgstation/tgstation/commit/99ed48ce8d3ba7e2d26d68519e05eae277c94685

**Comment by SweptWasTaken** [comment]
https://github.com/tgstation/tgstation/commit/e09101cb4be860ff02e035b18bef93a08b9c204c

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**